### ISSUE TITLE
Spacevim interface collapses to a single line suddenly

### ISSUE THREAD

**Issue by programmador** [wsdjeg/SpaceVim#3544]
Interface is collapsing to a single line after accidenal executing some kind of key sequence.
3 lines actually: header+1 line of code + footer.
I don't know what key sequence causes it. Probably I'm forgetting in which mode I'm staying and for example trying to print smth in normal mode.
It happens few times a day.
I've googled and didn't find any ways to collapse/restore interface intentionally.
I don't know whether is is a bug or not and whether it relates to spacevim or neovim.
The only way to restore interface is to exit spacevim and start it again.

## Expected behavior, english is required
Nothing should happen. Though probably I do it myself accidently - then It would be nice to know what exactly am I doing but I can't get what keys cause this behaviour because every time it's unexpected.

## The reproduce ways from Vim starting (Required!)
I don't know how exactly to reproduce it. Some kind of hotkey sequence.

## Debug info

> Please press <kbd>SPC h I</kbd>, debug info will be put into clipboard, then paste all content below.

Actually not, it's not being put into clipboard while using nvim inside of a terminal emulator.
Tells me 'clipboard: no provider' if I try to copy myself. Strange, it was working some days ago after I've set clipboard to unnamedplus in autoload.
I don't know whether any debug info would be helpful because apparently I'm causing the described behaviour myself and it is not a bug.
Anyway I'm using neovim if it helps.

## Screenshots

![2020-05-19-17:01:51](https://user-images.githubusercontent.com/674995/82337474-7c051800-99f4-11ea-86eb-c6254cf8b837.png)

**Comment by programmador** [comment]
Fixed clipboard: no xclip or smth similar was installed

**Comment by wsdjeg** [comment]
This can not be reproduced.

**Comment by programmador** [comment]
Unfortunately yes, it cannot be reproduced, even I don't know how does it happen.
It's always smth different and recently it happens quite rarely - I can't remember when I've encountered it for the last time.
Probably it should be closed if nobody else encounters it.

**Comment by wsdjeg** [comment]
If it is not reproduceable, we can not fix it. feel free to reopen if you can reproduce it.

**Comment by programmador** [comment]
Lately I've been encountering the same quite often. It doesn't seem to be related to a specific key sequence.
The latest time it was just space -> escape (open and close menu).
But I've figured out how to fix it without reloading the whole vim.

If anybody encounters the same - just type `:set winheight=999`.

Actually for me also nerdtree width was changed significantly so I also had to do smth like:
`100` -> `^w` -> `<`
while the focus is in nerdtree's buffer.

**Comment by programmador** [comment]
Though after the fix:
1. lsp popups are in strange places. They take a lot of whitespace. Though most of them are still readable.
2. rgrep search results are now fullscreen.
Probably 999 was too much.

**Comment by wsdjeg** [comment]
But we do not change the `winheight` option in spacevim. it should be done by some plugin.

**Comment by programmador** [comment]
Here is my config with plugins.
I can only say that it is definitely not a sidebar. Collapsing occurs with any sidebar - vimfiler, defx, nerdtree.

```
#=============================================================================
# basic.toml --- basic configuration example for SpaceVim
# Copyright (c) 2016-2017 Wang Shidong & Contributors
# Author: Wang Shidong < wsdjeg at 163.com >
# URL: https://spacevim.org
# License: GPLv3
#=============================================================================

[options]
    # set spacevim theme. by default colorscheme layer is not loaded,
    # if you want to use more colorscheme, please load the colorscheme
    # layer
    colorscheme = "one"
    colorscheme_bg = "light"
    # Disable guicolors in basic mode, many terminal do not support 24bit
    # true colors
    enable_guicolors = true
    # Disable statusline separator, if you want to use other value, please
    # install nerd fonts
    statusline_separator = "nil"
    statusline_inactive_separator = "bar"
    buffer_index_type = 4
    windows_index_type = 3
    enable_tabline_filetype_icon = false
    enable_statusline_mode = false
    statusline_unicode_symbols = false
    # Enable vim compatible mode, avoid changing origin vim key bindings
    vimcompatible = true
    autocomplete_method = "coc"
    default_indent = 4
    filetree_direction = "left"
    sidebar_width = 60
    enable_vimfiler_gitstatus = true
    bootstrap_before = "myspacevim#before"
    filemanager = "nerdtree"
    #filemanager = "defx"
    #enable_vimfiler_filetypeicon = false
    #disabled_plugins = ['defx-icons']

[[layers]]
    name = 'autocomplete'
    auto_completion_return_key_behavior = "complete"
    auto_completion_tab_key_behavior = "cycle"

[[layers]]
    name = "lsp"
    filetypes = [
        "dart",
        "php"
    ]
    [layers.override_cmd]
      php = ["intelephense", "--stdio"]
      dart = ["dart" "/opt/dart-sdk/bin/snapshots/analysis_server.dart.snapshot", "--lsp"]

[[layers]]
    name = "colorscheme"

[[layers]]
    name = "ctrlspace"
    enable-spacevim-styled-keys = true

[[layers]]
    name = "autocomplete"

[[layers]]
    name = "VersionControl"

[[layers]]
    name = "git"
    git-plugin = "fugitive"

[[layers]]
    name= "gtags"

[[layers]]
    name = "sudo"

[[layers]]
    name = 'shell'
    default_position = 'top'
    default_height = 30

[[layers]]
    name = "lang#dart"

[[layers]]
    name = "lang#php"

```

**Comment by programmador** [comment]
Hmmm, currently that variable is not set at all.
I'll try to call `let`or `echo` next time it happens to determine whether it has been set.
The fact that interface could be fixed with winheight does not mean that it has been broken with winheight.

**Comment by programmador** [comment]
It looks like I've found what could cause the issue.
I'm using xmonad and alacritty and interface collapsing could be related to them.
I'm also using a picom compositor which could also cause it.

I've encountered it once more and:
1. Opened a new window/tile in a new column. Nothing happened
2. Switched to line layout instead of column layout and it fixed vim's interface.

So it seems like for example alacritty tells vim that it's height has changed in some specific moment.
Maybe it's still related to vim itself and not to it's environment - I don't know.

BTW a similar issue could be encountered from time to time if You try to run any kind of TUI with docker - no matter whether it's vim, emacs, tmux or anything else. If You include for example vim/emacs in docker's startup command - interface will be collapsed to some "default" size (for me it's usually 1/4 of screen) until You try to resize the terminal emulator window.
If You leave just a pure shell as docker's startup command and run vim/emacs manually from the shell after the container gave you tty control - then interface will already have full size at the start moment.

So definitely there could be moments when shell does not tell it's size to the wrapped program for some reason.

And the fact that nobody except me reported the same issue also tells that it's rather related to quite a rare environment setup than to vim itself.

I think the issue could now be definitely closed.

And for anybody who encounters smth similar I could just leave here an advice to try to resize the window. In case of using tiling window managers it could be done for example by opening a new window and switching through layout variants.
Exactly, making DE to change terminal's height fixes the issue immediately.
